src/lib/crypto/kimchi_bindings/js/js_backend/finite_field.js

```diff
@@ -0,0 +1,460 @@
+  if (x === undefined) return 0; // None
+  return [0, x]; // Some(x)
+}
+// Provides: caml_pasta_fq_option
+function caml_pasta_fq_option(x) {
+  if (x === undefined) return 0; // None
+  return [0, x]; // Some(x)
+}
+
+// Provides: caml_pasta_fp_inv
+// Requires: caml_finite_field_inverse, caml_pasta_p_bigint, caml_pasta_fp_option
+function caml_pasta_fp_inv(x) {
+  var xinv = [caml_finite_field_inverse(x[0], caml_pasta_p_bigint)];
+  return caml_pasta_fp_option(xinv);
+}
+// Provides: caml_pasta_fq_inv
+// Requires: caml_finite_field_inverse, caml_pasta_q_bigint, caml_pasta_fq_option
+function caml_pasta_fq_inv(x) {
+  var xinv = [caml_finite_field_inverse(x[0], caml_pasta_q_bigint)];
+  return caml_pasta_fq_option(xinv);
+}
+
+// Provides: caml_pasta_fp_div
+// Requires: caml_bigint_modulo, caml_finite_field_inverse, caml_pasta_p_bigint
+function caml_pasta_fp_div(x, y) {
+  return [caml_bigint_modulo(x[0] * caml_finite_field_inverse(y[0], caml_pasta_p_bigint), caml_pasta_p_bigint)];
+}
+// Provides: caml_pasta_fq_div
+// Requires: caml_bigint_modulo, caml_finite_field_inverse, caml_pasta_q_bigint
+function caml_pasta_fq_div(x, y) {
+  return [caml_bigint_modulo(x[0] * caml_finite_field_inverse(y[0], caml_pasta_q_bigint), caml_pasta_q_bigint)];
+}
+
+// Provides: caml_pasta_fp_square
+// Requires: caml_pasta_fp_mul
+function caml_pasta_fp_square(x) {
+  return caml_pasta_fp_mul(x, x);
+}
+// Provides: caml_pasta_fq_square
+// Requires: caml_pasta_fq_mul
+function caml_pasta_fq_square(x) {
+  return caml_pasta_fq_mul(x, x);
+}
+
+// Provides: caml_pasta_fp_is_square
+// Requires: caml_finite_field_is_square, caml_pasta_p_bigint, caml_js_to_bool
+function caml_pasta_fp_is_square(x) {
+  var is_square = caml_finite_field_is_square(x[0], caml_pasta_p_bigint);
+  return caml_js_to_bool(is_square);
+}
+// Provides: caml_pasta_fq_is_square
+// Requires: caml_finite_field_is_square, caml_pasta_q_bigint, caml_js_to_bool
+function caml_pasta_fq_is_square(x) {
+  var is_square = caml_finite_field_is_square(x[0], caml_pasta_q_bigint);
+  return caml_js_to_bool(is_square);
+}
+
+// Provides: caml_pasta_fp_sqrt
+// Requires: caml_finite_field_sqrt, caml_pasta_fp_option, caml_pasta_p_bigint, caml_pasta_pm1_odd_factor, caml_twoadic_root_fp
+function caml_pasta_fp_sqrt(x) {
+  var sqrt = [caml_finite_field_sqrt(x[0], caml_pasta_p_bigint, caml_pasta_pm1_odd_factor, caml_twoadic_root_fp)];
+  return caml_pasta_fp_option(sqrt);
+}
+// Provides: caml_pasta_fq_sqrt
+// Requires: caml_finite_field_sqrt, caml_pasta_fq_option, caml_pasta_q_bigint, caml_pasta_qm1_odd_factor, caml_twoadic_root_fq
+function caml_pasta_fq_sqrt(x) {
+  var sqrt = [caml_finite_field_sqrt(x[0], caml_pasta_q_bigint, caml_pasta_qm1_odd_factor, caml_twoadic_root_fq)];
+  return caml_pasta_fq_option(sqrt);
+}
+
+// Provides: caml_pasta_fp_equal
+// Requires: caml_pasta_fp_sub, _0n
+function caml_pasta_fp_equal(x, y) {
+  return Number(caml_pasta_fp_sub(x, y)[0] === _0n);
```

Comment:
Can we switch to a direct `===`?

Reply:
yeah. I guess the idea was to test equality modulo p.. but right now we keep the numbers within `[0, p)` anyway so shouldn't matter